File: AndroidStudioProjects/LoginPage/app/src/main/java/com/example/loginpage/MainActivity.java
package com.example.loginpage;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private EditText text1;
    private EditText text2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button b1 = findViewById(R.id.button1);
        Button b2 = findViewById(R.id.button2);
        text1 = findViewById(R.id.editText1);
        text2 = findViewById(R.id.editText2);

        String id = text1.getText().toString();
        String pass = text2.getText().toString();

        b1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                text1.setText(null);
                text2.setText(null);
                Toast.makeText(MainActivity.this, "Logged in", Toast.LENGTH_LONG).show();
            }
        });

        b2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                text1.setText(null);
                text2.setText(null);
                Toast.makeText(MainActivity.this,"Log in cancelled",Toast.LENGTH_LONG).show();
            }
        });
    }
}

File: Assignment/Splash/app/src/main/java/com/example/splash/MainActivity.kt
package com.example.splash

import android.content.Intent
import android.content.SharedPreferences
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.EditText
import android.widget.Toast

class MainActivity : AppCompatActivity() {

    private lateinit var username: EditText
    private lateinit var password: EditText
    private lateinit var login: Button
    private lateinit var cancel: Button
    private val TAG = "Main Activity"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        username = findViewById(R.id.ed_username)
        password = findViewById(R.id.ed_password)
        login = findViewById(R.id.bt_login)
        cancel = findViewById(R.id.bt_cancel)

        login.setOnClickListener {
            var userDetails = username.text.toString()
            var passDetails = password.text.toString()

            if(userDetails.length >= 3 && passDetails.length >= 3) {
                var sP: SharedPreferences = getSharedPreferences("LoginDetails", MODE_PRIVATE)
                var editor: SharedPreferences.Editor = sP.edit()
                editor.putString("userDetails", userDetails)
                editor.putString("passDetails", passDetails)
                editor.apply()

                var intent = Intent(this,SecondActivity::class.java)
                startActivity(intent)
            } else {
                Toast.makeText(this,"Enter valid Credentials",Toast.LENGTH_LONG).show()
            }

        }

    }
}